File: dp/TypeScript/ads/sortList/drill/one.py
lista = [64, 34, 25, 12, 22, 11, 90]


def bubbleSort(lista):
    p1=0
    lista_acumulativa = []
    #import pdb;pdb.set_trace()
    while p1 < len(lista)-1:
        p2=p1+1
        lista_acumulativa.append(lista)
        while p2 < len(lista):
            if lista[p1]>lista[p2]:
                lista[p1],lista[p2] = lista[p2],lista[p1]
                p2+=1
                lista_acumulativa.append(lista)
                break
            p2=0
            p1+=1

    return lista


lista = [64, 34, 25, 12, 22, 11, 90]
print(bubbleSort(lista))

File: dp/TypeScript/ads/sortList/drill/one.js
let lista = [
  22, -33, 44, 55, -66, 0, 30, 12, 894, 2, 1223, 5, 32, 11, -12332, 2,
];

function bubbleSort(lista) {
  let p1 = 0;
  let len = lista.length;

  while (p1 < len - 1) {
    let p2 = p1 + 1;
    while (p2 < len) {
      let v1 = lista[p1];
      let v2 = lista[p2];
      let temp = v1;
      if (v1 > v2) {
        lista[p1] = v2;
        lista[p2] = temp;
      }
      p2 = p2 + 1;
    }
    p1 = p1 + 1;
  }
  return lista;
}

console.log(bubbleSort(lista));
